x/concentrated-liquidity/incentives.go

```diff
@@ -415,3 +415,86 @@ func (k Keeper) initOrUpdatePositionUptime(ctx sdk.Context, poolId uint64, posit
 
 	return nil
 }
+
+// collectIncentives collects incentives for all uptime accumulators for all positions belonging to `owner` in the given range.
+// Upon successful collection, it bank sends the incentives from the pool address to the owner and returns the collected coins.
+// Returns error if:
+// - pool with the given id does not exist
+// - no position given by pool id, owner, lower tick and upper tick exists
+// - other internal database or math errors.
+func (k Keeper) collectIncentives(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick int64, upperTick int64) (sdk.Coins, error) {
```

Comment:
This functions logic is very similar to the collectFees logic. Maybe this should be refactored to prevent code duplication between the two.

As a side note, I expect we will eventually be making a function that calls both collectIncentives and collectFees at the same time, since I can't think of a reason someone would one to collect one and not the other.

Reply:
I agree that this is a path worth considering, although tying the infrequent and more expensive operation of incentive claiming with the frequent and cheaper fee claiming would add an unnecessary amount of overhead for active market makers.

I do think there's a case for keeping them separate, even if just to save gas!

Reply:
Agreed on having the ability to call them separately! I still think the logic can be abstracted into a helper func though since they both utilize similar logic

Reply:
Agreed & discussed offline to move to separate issue to not block this PR #4514